packages/data/src/components/use-select/test/index.js

```diff
@@ -434,6 +434,63 @@ describe( 'useSelect', () => {
 			);
 		} );
 
+		it( 'captures state changes before the subscription is set', () => {
```

Comment:
@jsnajdr I am struggling to identify a concise way to describe this test based upon your findings. What do you think is an appropriate description for this test?

Reply:
I would say something like "captures state changes scheduled between render and effect." The key part of `useSelect` here is that it reads a ref in render and writes it in effect:
```js
function useSelect() {
  const value = ref.current;
  useEffect( () => {
    ref.current = value;
  } );
}
```
And then there is some `Inner` component that writes the ref in effect, too:
```js
function Inner() {
  useEffect( () => {
    ref.current = newValue;
  } );
  return null;
}
```
Now when you combine them together:
```js
function Outer() {
  useSelect();
  return <Inner />;
}
```
the effects are executed in children first and in parents second, so the order of operations is:
1. outer render reads
2. inner effect writes new value
3. outer effect writes stale value

In reality the `useSelect` effects are layout effects and the `Inner` component uses `componentDidUpdate`, but it leads to the same scheduling.